lib/matplotlib/axis.py

```diff
@@ -2081,8 +2081,9 @@ def _update_label_position(self, renderer):
         if self.label_position == 'bottom':
             try:
                 spine = self.axes.spines['bottom']
-                spinebbox = spine.get_transform().transform_path(
-                    spine.get_path()).get_extents()
```

Comment:
OK, so this is actually a behaviour change on bare axes.  OTOH  I think it is correct. `get_path()` will not include the ticks, whereas `get_window_extent` does.  For axes with labels, this doesn't matter because the tick labels dominate.  However, for an axes with no tick labels and long ticks, this spacing will be wrong.  

Apparently we only test this in one spot, because it is a bit odd to label an axes with not tick labels...

Reply:
Note in the changed image how the "ylabel" on the second axes has moved to the left a little bit.

Not sure if this is worth an API change note...